src/System.Slices/src/System/MemoryUtils.Other.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Runtime.InteropServices;
+
+namespace System
+{
+    internal static partial class MemoryUtils
+    {
+        /// <summary>
+        /// platform independent fast memory comparison based on idea from String.EqualsHelper 
+        /// </summary>
+        internal static bool MemCmp<[Primitive]T>(Span<T> first, Span<T> second)
+            where T : struct
+        {
+            unsafe
+            {
+                // prevent GC from moving memory
+                var firstPinnedHandle = GCHandle.Alloc(first.Object, GCHandleType.Pinned);
+                var secondPinnedHandle = GCHandle.Alloc(second.Object, GCHandleType.Pinned);
+
+                try
+                {
+                    byte* firstPointer = (byte*)PtrUtils.ComputeAddress(first.Object, first.Offset).ToPointer();
+                    byte* secondPointer = (byte*)PtrUtils.ComputeAddress(second.Object, second.Offset).ToPointer();
+
+                    long bytesCount = first.Length * PtrUtils.SizeOf<T>();
+
```

Comment:
perhaps it is better to avoid long here. We can compute an upper limit for one of the ptrs and compare against that.